### PR TITLE
Revision of string interpolation quickstart

### PR DESCRIPTION
**Pull request by pkulikov**
On the journey related to #3285 I've revised the string interpolation [quickstart](https://docs.microsoft.com/en-us/dotnet/csharp/quick-starts/interpolated-strings). The philosophy behind made changes is the following.

First, I use *string interpolation* when referring to the feature and *interpolated string* when talking about a string literal. That looks that folks prefer *string interpolation* term when [searching in Google](https://trends.google.com/trends/explore?q=c%23%20string%20interpolation,c%23%20interpolated%20string). Let's hope those changes will bring the quickstart page to the top of the search results.

Second, [current comments](https://docs.microsoft.com/en-us/dotnet/csharp/quick-starts/interpolated-strings#step-5) under the quickstart show that audience is confused. That's why I've tried to simplify the quickstart, especially its second step.

At last, I've made one more pass on the [feature description itself](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated) to make it smoother.

@BillWagner @rpetrusha please review.

### CODE REVIEW

File: docs/csharp/quick-starts/index.md [hunk 1]
```diff
@@ -18,20 +18,20 @@ that you can run in your browser.
...
-the online browser experience or on your own development
-environment. At the end of each lesson, you decide if you want to continue
+the online browser experience or [in your own local development
+environment](local-environment.md). At the end of each quickstart, you decide if you want to continue
 with the next quickstart online or on your own machine. There are links
 to help you setup your environment and continue with the next quickstart
 on your machine.
 
-## [Hello world](hello-world.yml)
+## [Hello World](hello-world.yml)
```
Comment (BillWagner) [inline docs/csharp/quick-starts/index.md]: Headings should stay sentence cased, not title cased.

File: docs/csharp/quick-starts/index.md [hunk 2]
```diff
@@ -18,20 +18,20 @@ that you can run in your browser.
...
 
-In the [Hello world](hello-world.yml) quickstart, you'll create the most basic
+In the [Hello World](hello-world.yml) quickstart, you'll create the most basic
```
Comment (BillWagner) [inline docs/csharp/quick-starts/index.md]: Same: "Hello world" is correct.

File: docs/csharp/quick-starts/index.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ numeric types. You'll learn the basics of rounding, and how to perform
 mathematical calculations using C#. This quickstart is also available
 [to run locally on your machine](numbers-in-csharp-local.md).
 
-This quickstart assumes that you have finished the [Hello world](hello-world.yml) tutorial.
+This quickstart assumes that you have finished the [Hello World](hello-world.yml) lesson.
```
Comment (BillWagner) [inline docs/csharp/quick-starts/index.md]: Hello world.

File: docs/csharp/quick-starts/interpolated-strings-local.md [hunk 4]
```diff
@@ -1,52 +1,65 @@
...
 
 ```console
-dotnet new console -n interpolated -o .
+dotnet new console
 ```
 
 This command creates a new .NET Core console application in the current directory.
 
-Open **Program.cs** in your favorite editor, and replace the line `Console.WriteLine("Hello World!");` with the following code, where you replace `<name>` with your name:
+Open **Program.cs** in your favourite editor, and replace the line `Console.WriteLine("Hello World!");` with the following code, where you replace `<name>` with your name:
```
Comment (BillWagner) [inline docs/csharp/quick-starts/interpolated-strings-local.md]: Our standard uses the American spellings (favorite).

File: docs/csharp/quick-starts/interpolated-strings-local.md [hunk 5]
```diff
@@ -60,49 +73,46 @@ public class Vegetable
...
-   public enum Unit { item, pound, ounce, dozen };
-   
    public static void Main()
    {
       var item = new Vegetable("eggplant");
       var date = DateTime.Now;
       var price = 1.99m;
-      var unit = Unit.item;
-      Console.WriteLine($"On {date}, the price of {item} was {price} per {unit}.");
+      Console.WriteLine($"On {date}, the price of {item} was {price} per item.");
```
Comment (BillWagner) [inline docs/csharp/quick-starts/interpolated-strings-local.md]: I do like the example that adds the `enum` field value. That's useful for new developers.

File: docs/csharp/quick-starts/interpolated-strings-local.md [hunk 6]
```diff
@@ -60,49 +73,46 @@ public class Vegetable
...
 
-- If the interpolated expression is not `null`, the `ToString` method of the type of the interpolated expression is called. You can test this by commenting out the definition of the `Vegetable.ToString` method in the example by putting a comment symbol (`//`) in front of it. In the output, the string "eggplant" is replaced by the type name, "Vegetable", which is the default behavior of the <xref:System.Object.ToString?displayProperty=nameWithType> method.   
+- If the interpolated expression doesn't evaluate to `null`, typically the `ToString` method of the result type is called. You can test this by updating the implementation of the `Vegetable.ToString` method. You might even not implement `ToString` method since every C# data type has some implementation of this method. To test that, comment out the definition of the `Vegetable.ToString` method in the example (to do that, put a comment symbol `//` in front of it). In the output, the string "eggplant" is replaced by the fully qualified type name ("Vegetable" in this example), which is the default behavior of the <xref:System.Object.ToString?displayProperty=nameWithType> method.
```
Comment (BillWagner) [inline docs/csharp/quick-starts/interpolated-strings-local.md]: An additional sentence here describing the default behavior for enum fields would be useful.

File: docs/csharp/quick-starts/interpolated-strings-local.md [hunk 7]
```diff
@@ -60,49 +73,46 @@ public class Vegetable
...
+In the previous section, two poorly formatted strings were inserted into the result string. One was a date and time value for which only the date was appropriate. The second was a price that didn't indicate its unit of currency. Both issues are easy to address. String interpolation lets you specify *format strings* that control the formatting of particular types. Modify the call to `Console.WriteLine` from the previous example to include the format strings for the date and price expressions as shown in the following line:
 
 ```csharp
-Console.WriteLine($"On {date:d}, the price of {item} was {price:C2} per {unit}.");
+Console.WriteLine($"On {date:d}, the price of {item} was {price:C2} per item.");
```
Comment (BillWagner) [inline docs/csharp/quick-starts/interpolated-strings-local.md]: I'd like to revert to the previous sample.

File: docs/csharp/quick-starts/interpolated-strings-local.md [hunk 8]
```diff
@@ -111,54 +121,48 @@ public class Example
 {
    public static void Main()
    {
-      var titles = new Dictionary<string, string>();
-      titles.Add("Doyle, Arthur Conan", "Hound of the Baskervilles, The");
-      titles.Add("London, Jack", "Call of the Wild, The");
-      titles.Add("Shakespeare, William", "Tempest, The");
+      var titles = new Dictionary<string, string>()
```
Comment (BillWagner) [inline docs/csharp/quick-starts/interpolated-strings-local.md]: Great addition.  😎

File: docs/csharp/quick-starts/interpolated-strings.yml [hunk 9]
```diff
@@ -55,97 +55,80 @@ items:
...
-          var item = new Vegetable("eggplant");
-          var date = DateTime.Now;
-          var price = 1.99m;
-          var unit = Unit.item;
-          Console.WriteLine($"On {date}, the price of {item} was {price} per {unit}.");
-       }
-    }
+    var item = new Vegetable("eggplant");
+    var date = DateTime.Now;
+    var price = 1.99m;
```
Comment (BillWagner) [inline docs/csharp/quick-starts/interpolated-strings.yml]: I like the addition of the enum example here.

File: docs/csharp/quick-starts/interpolated-strings.yml [hunk 10]
```diff
@@ -55,97 +55,80 @@ items:
...
+    - If the interpolated expression doesn't evaluate to `null`, typically the `ToString` method of the result type is called. You can test this by updating the implementation of the `Vegetable.ToString` method. You might even not implement `ToString` method since every C# data type has some implementation of this method. To test that, comment out the definition of the `Vegetable.ToString` method in the example (to do that, put a comment symbol `//` in front of it). In the output, the string "eggplant" is replaced by the fully qualified type name (the namespace defined by the C# REPL along with the type name), which is the default behavior of the <xref:System.Object.ToString?displayProperty=nameWithType> method.
```
Comment (BillWagner) [inline docs/csharp/quick-starts/interpolated-strings.yml]: As before, an additional statement about the enum behavior would be helpful.

File: docs/csharp/quick-starts/interpolated-strings.yml [hunk 11]
```diff
@@ -55,97 +55,80 @@ items:
...
+    In the previous step, two poorly formatted strings were inserted into the result string. One was a date and time value for which only the date was appropriate. The second was a price that didn't indicate its unit of currency. Both issues are easy to address. String interpolation lets you specify *format strings* that control the formatting of particular types. Modify the call to `Console.WriteLine` from the previous example to include the format strings for the date and price expressions as shown in the following line:
 
     ```csharp
-    Console.WriteLine($"On {date:d}, the price of {item} was {price:C2} per {unit}.");
+    Console.WriteLine($"On {date:d}, the price of {item} was {price:C2} per item.");
```
Comment (BillWagner) [inline docs/csharp/quick-starts/interpolated-strings.yml]: As before, I prefer the previous example with the enum field.

File: docs/csharp/quick-starts/interpolated-strings.yml [hunk 12]
```diff
@@ -55,97 +55,80 @@ items:
...
-      var inventory = new Dictionary<string, int>();
-      inventory.Add("hammer, ball pein", 18);
-      inventory.Add("hammer, cross pein", 5);
-      inventory.Add("screwdriver, Phillips #2", 14);
+      var inventory = new Dictionary<string, int>()
+      {
+          ["hammer, ball pein"] = 18,
+          ["hammer, cross pein"] = 5,
+          ["screwdriver, Phillips #2"] = 14
+      };
```
Comment (BillWagner) [inline docs/csharp/quick-starts/interpolated-strings.yml]: 😎 awesome.

File: docs/csharp/quick-starts/local-environment.md [hunk 13]
```diff
@@ -39,7 +39,7 @@ how computers store numbers and how to perform calculations with different
 numeric types. You'll learn the basics of rounding, and how to perform
 mathematical calculations using C#. 
 
-This quickstart assumes that you have finished the [Hello world](hello-world.yml) tutorial.
+This quickstart assumes that you have finished the [Hello World](hello-world.yml) lesson.
```
Comment (BillWagner) [inline docs/csharp/quick-starts/local-environment.md]: "World" should be lower case.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Overall, this is great work @pkulikov.  

I had a few comments. Most are pertaining to dropping the use of an `enum` field in the sample. That's a very useful concept for beginners.

**Comment by pkulikov** [comment]
@BillWagner thank you for the feedback, I've addressed all comments.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks for making the updates @pkulikov It looks great.  I've approved the changes, and I'll :shipit: now.

Thanks again for your continued involvement.

**Comment by pkulikov** [comment]
@BillWagner thank you very much for [the highlight](https://docs.microsoft.com/en-us/teamblog/release-notes-april-2018#contributor-highlights). 
So far, it's been very cool to work with you and your team (@mairaw @rpetrusha you also rock): your reviews and feedback are always good and welcoming; of course, my favorite one is :shipit: :).

**Pull request merged**